### ISSUE TITLE
[Security] Android keyboard app does store the password entered in Mattermost App for sign-in

### ISSUE THREAD

**Issue by c0deright** [mattermost/mattermost-mobile#7137]
#### Summary

On android when using Mattermost App to sign into a server the keyboard stores (parts of) the password and uses it for autocomplete/suggest later on.

### Environment Information

  - Device Name: Xiaomi Redmi K20 Pro
  - OS Version: Android 9
  - Mattermost App Version: 2.0.1 (Build 6000456)
  - Mattermost Server Version: 7.7.1
  - Keyboard: SwiftKey 8.10.32.13

#### Steps to reproduce

  - Use a generated, complex password like `Hp!JTqSzp\SLm%KA9xhw` for your mattermost user on any given mattermost server.
  - Install Mattermost Android App and sign into your server
  - Wipe Mattermost Android App Data and open Mattermost Android App again
  - Mattermost will prompt you to add a new server and sign into it
  - When typing parts of the password SwiftKey Keyboard will suggest the next part up to the next special character, e.g. you type `Hp!` and `JTqSzp` will appear in the keyboards suggest row (this is just an example and might not work 100% but Swiftkey did just suggest parts of my password after uninstalling and reinstalling Mattermost Android App)
  - this did not happen to me on any other app before - Mattermost being the 1st time I did notice Swiftkey did store a previously entered password

#### Expected behavior

The keyboard app does NOT store parts of the password in the 1st place nor does it suggest parts of it.

#### Observed behavior (that appears unintentional)

The keyboard app did save parts of the password and suggested parts when it noticed I was typing my password again.

#### Possible fixes

Set flags for the password field so keyboard apps know not to store entered characters (I have no idea if and how this works on android).

**Comment by amyblais** [comment]
@c0deright Thank you, would you be open to sharing these reports via responsibledisclosure@mattermost.com? Please see more on our Responsible Disclosure Policy here https://mattermost.com/security-vulnerability-report/.

**Comment by enahum** [comment]
Have you tried any other keyboard? Gboard maybe? Cause there is nothing we do in the app to store anything on any keyboard, so I just want to confirm

**Comment by c0deright** [comment]
Ok, I did dig a bit deeper and it seems to be an issue outside of Mattermost, I think.

The password field has a eye symbol to the right that is crossed out by default, resulting in entered characters to be replaced with `•`. In this mode your 5 chars long password looks like `•••••` and the keyboard **does not save** the string - everything is fine.

When the crossed out eye symbol is tapped, the characters entered aren't replaced by `•` and the keyboard **does save** the string - this is the issue I stumbled over because of my complicated and long password I wanted to make sure to enter correctly.

I guess it's more of an Android thing then. Using Android 9.
